homework/grayscale-image/compression.cpp

```diff
@@ -0,0 +1,62 @@
+#include "compression.hpp"
+
+#include <iostream>
+#include <algorithm>
+
+std::vector<std::pair<uint8_t, uint8_t>> compressGrayscale(std::array<std::array<uint8_t, width>, height> &compress_array)
+{
+
+    std::vector<std::pair<uint8_t, uint8_t>> vec;
+    vec.reserve(width * height);
+
+    std::for_each(compress_array.cbegin(), compress_array.cend(), [&vec](auto el)
```

Comment:
for_each for me is similar to for. You can use transform :) or std::generate (you will capture iterator and then jump between different numbers and return proper pair :) )